jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/ByteBufferContentSource.java

```diff
@@ -78,6 +78,8 @@ public Content.Chunk read()
             if (last)
                 terminated = Content.Chunk.EOF;
         }
+        if (!buffer.hasRemaining())
```

Comment:
We should probably javadoc that the code instantiating this class is responsible for managing the lifecycle of the buffers, unlike `ChunksContentSource` for which the exact opposite is true.